tower frontend:unprivileged env vars docs (#354)
seqeralabs/nf-tower-docs: docs/enterprise/kubernetes.md

kubectl apply -f tower-svc.yml
```

> Note that an unprivileged version of the Tower frontend image is also available, which listens on

Reviewer: ```suggestion
Note that an unprivileged version of the Tower frontend image is also available. This image listens on
```

> an unprivileged port and therefore doesn't need to run as the root user.

Reviewer: ```suggestion
an unprivileged port and therefore doesn't need to be run as the root user.
```

> Replace the tag of the frontend image `cr.seqera.io/private/nf-tower-enterprise/frontend:v23.1.0`

Reviewer: ```suggestion
Replace the tag of the frontend image `cr.seqera.io/private/nf-tower-enterprise/frontend:v23.x.x`
```

> with `cr.seqera.io/private/nf-tower-enterprise/frontend:v23.1.0-unprivileged`, and in the

Reviewer: ```suggestion
with `cr.seqera.io/private/nf-tower-enterprise/frontend:v23.x.x-unprivileged`. In the
```

> `frontend` service below specify the `targetPort` to match the one defined with the environment

Reviewer: ```suggestion
`frontend` service below, specify the `targetPort` to match the environment
```

> variable `NGINX_LISTEN_PORT` (see below), like so:

Reviewer: ```suggestion
variable `NGINX_LISTEN_PORT` (see below):
```

```yaml
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: frontend
  labels:
    app: frontend
spec:
  ...
      containers:
        - name: frontend

>           image: cr.seqera.io/private/nf-tower-enterprise/frontend:v23.1.0-unprivileged

Reviewer: ```suggestion
          image: cr.seqera.io/private/nf-tower-enterprise/frontend:v23.x.x-unprivileged
```

          env:
            - name: NGINX_LISTEN_PORT  # If not defined, defaults to 8080.
              value: 8081
---
apiVersion: v1
kind: Service
metadata:
  name: frontend
spec:
  type: LoadBalancer
  ports:
    - port: 80
      containerPort: 8081
  selector:
    app: "frontend"
```

The new unprivileged image will soon deprecate the current image that requires root, and can be

> easily customized, e.g. some options can be defined using environment variables:

Reviewer: ```suggestion
easily customized, e.g., some options can be defined using environment variables:
```

- `NGINX_LISTEN_PORT` (default `8000`): to specify the port the nginx process will listen on

>   inside the container; the `containerPort` on the `frontend` service must match with the value

Reviewer: ```suggestion
  inside the container; the `containerPort` on the `frontend` service must match the value
```

>   defined for the environment variable.

Reviewer: ```suggestion
  defined in the environment variable.
```

- `NGINX_LISTEN_PORT_IPV6` (default `8000`): same as `NGINX_LISTEN_PORT`, but for the port to open
  on the IPv6 address.
- `NGINX_UPSTREAM_HOST` (default `backend`): the hostname of the backend service that the nginx
  process will route requests to.
- `NGINX_UPSTREAM_PORT` (default `8080`): the port that the backend service is exposed to.

If further customizations of the config file are needed, mount a ConfigMap/Secret over the

> templated nginx configuration file at `/etc/nginx/templates/tower.conf.template`: see the

Reviewer: ```suggestion
templated nginx configuration file at `/etc/nginx/templates/tower.conf.template`: see
```
